### ISSUE TITLE
Port forwarding does not work on all ports

### ISSUE THREAD

**Issue by DLesage25** [coder/coder#7563]
My coder instance lives on an AWS EC2 using AL2. I've set up my security group to allow traffic from all origins on HTTP, HTTPS, SSL, TCP 7474 and TCP 7687. 

I have a workspace running with two applications, one running on port 3000 and the other on port 7474. When I try to set up port forwarding to port 3000, the URL works ok. If I do the same for port 7474, I get a `404 - Page Not Found` and the URL resolves to `https://[coder_host]/browser/`. 

What's weird is if I try to set up port forwarding to, say, port 8080, I get a 502. But that doesn't happen with port `7474`. 

I also set up a `coder_app` with url `http://localhost:7474/`, but I still get a 404 if I try to open from the workspace button.

**Comment by ericpaulsen** [comment]
hi @DLesage25 - can you confirm your `CODER_ACCESS_URL` and `CODER_WILDCARD_ACCESS_URL` values, and double check what is listening on `7474` via this command?

```
sudo lsof -i -P | grep LISTEN | grep :7474
```

**Comment by DLesage25** [comment]
Thanks @ericpaulsen! My `CODER_ACCESS_URL` and  `CODER_WILDCARD_ACCESS_URL` are set to the following values:

```
CODER_ACCESS_URL=https://coder.mydomain.com
CODER_WILDCARD_ACCESS_URL=*.coder.mydomain.com
```

Other ports' access URLs work just fine, so I think that's not the issue.

When running `sudo lsof -i -P | grep LISTEN | grep :7474` I get:
`docker-pr 2288      root    4u  IPv4 1286692      0t0  TCP *:7474 (LISTEN)`

If it helps, I set up sysbox to run docker-in-docker, and the container listed there is a DB image set up as part of my workspace. There's a docker-compose file that is cloned on init, and my startup script just runs `sudo docker compose up`. For what it's worth, I can connect to the DB from within the workspace.

**Comment by ericpaulsen** [comment]
@DLesage25 thanks for the added detail - so to confirm, the process running on port `7474` is a database running as a docker container inside of the workspace?

if that is the case, is the process running on `3000` also its own respective docker container?

**Comment by DLesage25** [comment]
@ericpaulsen that is correct. And the process running on `3000` is not a docker container. It's a nodeJS application that is not containerized. 

Something else I noticed - if my DB image (I'm using `neo4j:enterprise`, btw) is not running and I set up port forwarding, I get a 502, as I would expect. Once my image is running, I get a 404. 

I also tested running a base nginx container on port 3000 and setting up port forwarding and it worked. So the problem seems to be specific to the neo4j image. (?)

**Comment by DLesage25** [comment]
One last observation:

I tried accessing the workspace using vscode server and set up port forwarding from there. I am able to access `localhost:7474` and get redirected to `localhost:7474/browser` no problem, can access my DB interface. 

So this only seems to affect port forwarding set up from the dashboard AND trying to set up a coder_app that points to `localhost:7474`. Is there a way to configure Coder to work with those two?

**Comment by DLesage25** [comment]
@ericpaulsen that makes sense.  The application is initialized as part of the startup script, yes.

This is my `coder_app` configuration:
```
resource "coder_app" "neo4j" {
  agent_id  = coder_agent.main.id
  slug      = "neo4j"
  url       = "http://localhost:7474/browser"
  subdomain = false
  share     = "authenticated"

  healthcheck {
    url       = "http://localhost:7687"
    interval  = 5
    threshold = 10
  }

}
```

So presumably, if I remove the coder_app resource I should be able to set up port forwarding, correct? It's either/or in terms of `coder_app` and unique urls for port forwarding?

**Comment by ericpaulsen** [comment]
@DLesage25

i just tested on my own deployment, where I created a `coder_app` that listened on `6000`:

```
resource "coder_app" "test" {
  agent_id     = coder_agent.coder.id
  display_name = "test"
  icon         = "/icon/code.svg"
  url          = "http://localhost:6000/"
}
```

i then ran a python web server on `6000` and was able to get the application to open via _both_ the port-forwarded URL and the `coder_app` button.

however, tested by appending `/browser` to my `coder_app` URL (as you did above) and I was able to reproduce the 404. so it appears this `/browser` URL path is conflicting with the URL generated by the port-forward.

try removing `/browser` from the `coder_app` URL and test. so it isn't an issue of `coder_app` and port forward not being able to support the same application, it's just a matter of the URL parameter not being correct.
